__tests__/app.test.js

```diff
@@ -0,0 +1,164 @@
+      return request(app)
+        .get("/api/articles/not-an-id")
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
+        });
+    });
+    test("404 - given non existent id with no data, returns message: not found", () => {
+      return request(app)
+        .get("/api/articles/999999")
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("not found");
+        });
+    });
+  });
+
+  //ticket 7
+  describe("PATCH", () => {
+    test("200 - accepts body in form of { inc_votes: newVote } with newVote dictating the inc/decrement of votes. Responds with updated article", () => {
+      const votes = { inc_votes: -10 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(votes)
+        .expect(200)
+        .then((res) => {
+          expect(res.body.article).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: expect.any(String),
+            votes: 90,
+          });
+        });
+    });
+    //already covered by get request error?
+    test("400 - given invalid id data type, returns message: bad request", () => {
+      const votes = { inc_votes: -10 };
+      return request(app)
+        .patch("/api/articles/not-an-id")
+        .send(votes)
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
+        });
+    });
+    test("404 - given correct id with no data, returns message: not found", () => {
+      const votes = { inc_votes: -10 };
+      return request(app)
+        .patch("/api/articles/999999")
+        .send(votes)
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("not found");
+        });
+    });
+    //if no code change needed outside of test, is the error test redundant?
+    test(" 400- given incorrect votes data type returns message: bad request", () => {
+      const votes = { inc_votes: 'not-a-valid-vote-count' };
+      return request(app)
+        .patch("/api/articles/999999")
+        .send(votes)
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
+        });
+    });
+    test(" 400- given incorrect but existing key returns message: not found", () => {
+      const votes = { title: 9 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(votes)
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
+        });
+    });
+    test(" 400- given empty object returns message: bad request", () => {
+      const votes = {};
+      return request(app)
+        .patch("/api/articles/1")
+        .send(votes)
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
```

Comment:
Test - what if there are more than one key on the object? e.g. inc_votes and title. 

It should be handled already, but good to show its been considered.